journalpump/rsyslog.py

```diff
@@ -133,11 +134,26 @@ def close(self):
             self.socket = None
 
     def send(self, message):
-        if self.socket is None:
-            self._connect()
-        self.socket.sendall(message[:self.max_msg - 1])
-        if len(message) >= self.max_msg:
-            self.socket.sendall(b"\n")
+        for retry in [True, False]:
+            try:
+                if self.socket is None:
+                    self._connect()
+
+                self.socket.sendall(message[:self.max_msg - 1])
+                if len(message) >= self.max_msg:
+                    self.socket.sendall(b"\n")
+
+                break
+            except Exception as ex:  # pylint: disable=broad-except
+                self.close()
+
+                if not (retry and self._should_retry(ex=ex)):
```

Comment:
Nit: imo `not retry or not self._should_retry(ex=ex)` would read easier but very subjective.